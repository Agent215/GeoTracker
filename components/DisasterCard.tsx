import React from "react";
import { Card, CardItem, Text, Body } from "native-base";
import { Col, Grid } from "react-native-easy-grid";
import { useDispatch } from 'react-redux';
import { StyleSheet, Image, Share } from "react-native";
import * as actions from '../store/actions/actions';
import { useNavigation } from '@react-navigation/native';
import FeedScreenShare from '../components/ShareFeat';
import { IconButton } from "react-native-paper";
import DisasterIcon from '../components/CustomIcon';
import { accessibilityProps } from "react-native-paper/lib/typescript/src/components/MaterialCommunityIcon";
import { TouchableOpacity } from "react-native-gesture-handler";

// DisasterCard holds the layout to each feed card.
// The Card uses <Grid> to space objects out in header, and footer. 2/3 of header is description 1/3 is icon
const DisasterCard = (props) => {

  const dispatch = useDispatch();
  const navigation = useNavigation();

  const goToDisaster = (event) => {
    //set current disaster in store
    console.log(event.title)
    console.log(event.currentLat)
    console.log(event.currentLong)
    dispatch(actions.setCurrentDisaster(event));
    navigation.navigate("Map");
  };

  return (
    <Card>
      <CardItem header bordered>
        <Grid>
          <Col size={2}>
            <Text style={{ fontSize: 25 }}>
              {props.event.title}
            </Text>
          </Col>
          <Col style={styles.iconStyle}>
            <DisasterIcon 
              size={50}
              event={props.event}
            />
          </Col>
        </Grid>
      </CardItem>
      <CardItem footer>
        <Col>
          <FeedScreenShare sourceLink={props.event.sourceLink} color="black" size={50} />
          <Text style={styles.share}>Share</Text>
        </Col>
        <Col>

          <IconButton
            icon="crosshairs-gps"
            color={"black"}
            size={50}

            onPress={() => goToDisaster(props.event)}
          />
          <Text style={styles.go}>Go</Text>

        </Col>
        <Col>
          <TouchableOpacity
          onPress={() => { 
            dispatch(actions.unSaveDisaster(props.event));
          console.log("you are at line 71 of DisasterCard"); 
        }}
          >
            <IconButton
              icon="trash-can"
              color={"black"}
              size={50}
              
            />
          </TouchableOpacity>
          <Text style={styles.remove}>Remove</Text>
        </Col>
      </CardItem>
    </Card>
  );
}


const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    paddingHorizontal: 10
  },
  button: {
    alignItems: "center",
    backgroundColor: "white",
    padding: 0,
    margin: 0
  },
  countContainer: {
    alignItems: "center",
    padding: 10
  },
  share: {
    marginLeft: 25
  },
  go: {
    marginLeft: 33
  },
  remove: {
    marginLeft: 13
  },
  iconStyle: {
    marginLeft: 60,
    marginTop: 5
  
  }
});

export default DisasterCard;