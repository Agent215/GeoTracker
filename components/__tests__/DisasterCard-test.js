import * as React from 'react';
import renderer from 'react-test-renderer';

import DisasterCard from '../../components/DisasterCard';

// mock of a prop to pass disaster card
const mockProp =  { title: 'Thirsty Thursday', id: 1, description: 'Draught', LatL: 11, LongL: 12, icon: '../assets/Icons/Draught.png'};

/*
Check disaster card against snapshot. run jest -u to update snapshot
*/
it(`DisasterCard renders correctly`, () => {
  const tree = renderer.create(<DisasterCard event = {mockProp}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
/**
 * Disaster card should not be null
 */
it(`DisasterCard should not be null`, () => {
    const tree = renderer.create(<DisasterCard event = {mockProp}/>).toJSON();
  
    expect(tree).toBeTruthy();
  });
  
