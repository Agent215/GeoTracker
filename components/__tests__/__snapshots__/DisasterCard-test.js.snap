// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Disaster Card Component DisasterCard renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderBottomWidth": 0.5,
      "borderColor": "#ccc",
      "borderLeftWidth": 0.5,
      "borderRadius": 2,
      "borderRightWidth": 0.5,
      "borderTopWidth": 0.5,
      "elevation": 3,
      "flexWrap": "nowrap",
      "marginBottom": 5,
      "marginLeft": 2,
      "marginRight": 2,
      "marginTop": 5,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 1.5,
    }
  }
>
  <View
    bordered={true}
    header={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        size={2}
        style={
          Object {
            "flex": 2,
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#007aff",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Draught
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <Image
          source={1}
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
    >
      Thirsty Thursday
    </Text>
  </View>
  <View
    footer={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderBottomWidth": null,
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "margin": 6,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 37.5,
                "height": 75,
                "width": 75,
              },
              undefined,
              Object {
                "justifyContent": "center",
              },
            ],
          ]
        }
      >
        <View>
          <Text />
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        Share
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "margin": 6,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 37.5,
                "height": 75,
                "width": 75,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        <View>
          <Text />
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        Go
      </Text>
    </View>
  </View>
</View>
`;
